From 5c5cfb1e110a34c64f274af861316054070e1698 Mon Sep 17 00:00:00 2001
From: Frederik Schwan <frederik@tty42.de>
Date: Tue, 1 Nov 2022 18:14:03 +0100
Subject: [PATCH] Pin Java version

Inject JAVA_HOME and JAVA variables into scripts that end up in /usr/bin
to also pin the java version for the cli tools that otherwise would use
the default system JAVA_HOME and JAVA var.
---
 integration/client-cli/admin-cli/src/main/bin/kcadm.sh | 3 ++-
 quarkus/dist/src/main/content/bin/kc.sh                | 1 +
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/integration/client-cli/admin-cli/src/main/bin/kcadm.sh b/integration/client-cli/admin-cli/src/main/bin/kcadm.sh
index fa0e7f86a5..40505e414f 100755
--- a/integration/client-cli/admin-cli/src/main/bin/kcadm.sh
+++ b/integration/client-cli/admin-cli/src/main/bin/kcadm.sh
@@ -18,7 +18,8 @@ esac
 
 RESOLVED_NAME="${RESOLVED_NAME:-"$0"}"
 
-DIRNAME="$(dirname "$RESOLVED_NAME")"
+DIRNAME=`dirname "$RESOLVED_NAME"`
+. "$DIRNAME/common.sh"
 
 if [ -z "$JAVA" ]; then
     if [ -n "$JAVA_HOME" ]; then
diff --git a/quarkus/dist/src/main/content/bin/kc.sh b/quarkus/dist/src/main/content/bin/kc.sh
index 9a6e62e2dd..ed08bf2c2e 100644
--- a/quarkus/dist/src/main/content/bin/kc.sh
+++ b/quarkus/dist/src/main/content/bin/kc.sh
@@ -25,6 +25,7 @@ RESOLVED_NAME="${RESOLVED_NAME:-"$0"}"
 
 GREP="grep"
 DIRNAME="$(dirname "$RESOLVED_NAME")"
+. "$DIRNAME/common.sh"
 
 abs_path () {
   if [ -z "$IS_CYGWIN" ] ; then
-- 
2.46.0

